Project/Project/Controllers/HomeController.cs

```diff
@@ -1,20 +1,61 @@
 ﻿using System.Diagnostics;
 using Microsoft.AspNetCore.Mvc;
 using Project.Models;
+using Project.Data;
 
 namespace Project.Controllers;
 
 public class HomeController : Controller
 {
     private readonly ILogger<HomeController> _logger;
+    private readonly DataContext _context;
 
-    public HomeController(ILogger<HomeController> logger)
+    public HomeController(ILogger<HomeController> logger, DataContext context)
     {
         _logger = logger;
+        _context = context;
+    }
+    
+    public List<Rating> TopRated()
+    {
+        var movieIds = _context.Ratings.Select(r => r.Movie_id).Distinct().ToList();
+
+        var aggregatedRatings = new List<Rating>();
+
+        foreach (var movieId in movieIds)
+        {
+            var movieRatings = _context.Ratings.Where(r => r.Movie_id == movieId).ToList();
+
+            if (movieRatings.Count == 1)
+            {
+                aggregatedRatings.Add(movieRatings[0]);
+            }
+            else if (movieRatings.Count > 1)
+            {
+                var averageRating = new Rating
+                {
+                    Movie_id = movieId,
+                    Rating_value = Math.Round(movieRatings.Average(r => r.Rating_value),1),
+                    Movie_title = movieRatings[0].Movie_title,
+                    Movie_poster = movieRatings[0].Movie_poster,
+                    Created_at = DateTime.Now,
+                    Updated_at = DateTime.Now
+                };
+                aggregatedRatings.Add(averageRating);
+            }
+        }
+
+        var topRated = aggregatedRatings
+            .OrderByDescending(r => r.Rating_value)
+            .Take(10)
+            .ToList();
+
+        return topRated;
```

Comment:
Can you try if the code below works? As it shorter and simpler:
public List<Rating> TopRated()
{
    // Aggregating ratings and calculating the average rating for each movie
    var topRatedMovies = _context.Ratings
        .GroupBy(r => new { r.Movie_id, r.Movie_title, r.Movie_poster })
        .Select(g => new Rating
        {
            Movie_id = g.Key.Movie_id,
            Movie_title = g.Key.Movie_title,
            Movie_poster = g.Key.Movie_poster,
            Rating_value = Math.Round(g.Average(r => r.Rating_value), 1),
            Created_at = DateTime.Now,
            Updated_at = DateTime.Now
        })
        .OrderByDescending(r => r.Rating_value)
        .Take(10)
        .ToList();

    return topRatedMovies;
}